src/averageAges.js

```diff
@@ -56,6 +61,50 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  const filteredChildren = (!onlyWithSon)
+    ? people.filter(person => people
+      .find(mother => mother.name === person.mother))
+    : peopleFilter(
+      people.filter(person => people
+        .find(mother => mother.name === person.mother)),
+      'm'
+    );
```

Comment:
Better to write one filter method and write needed condition inside it instead of writing filter method for every case